beacon-chain/db/filesystem/blob.go

```diff
@@ -323,11 +325,14 @@ func (bs *BlobStorage) shouldPrune(slot primitives.Slot) bool {
 
 // pruneOlderThan prunes blobs in the base directory based on the retention epoch and current slot.
 func (bs *BlobStorage) pruneOlderThan(slot primitives.Slot) error {
-	err := bs.Prune(slot)
-	if err != nil {
-		return err
+	v := bs.atomicPrunedEpoch.CompareAndSwap(uint64(bs.lastPrunedEpoch), uint64(slots.ToEpoch(slot)))
+	if v {
+		bs.lastPrunedEpoch = slots.ToEpoch(slot)
```

Comment:
a bit of a code smell for the argument to be a slot, but you need it as an epoch. Consider changing the `slot` to an `epoch`, possibly in another PR

Reply:
ah nvm, i suppose you do prune by slot below.